### PR TITLE
Fix table export

### PR DESCRIPTION
**Pull request by A9G-Data-Droid**
Resolve bugs in Table Data export:
- Closes Issue #3
- Error in Export Query preparation
- Missing separators between fields in data rows
- Closes Issue #2
-  Solved bug in determining connect string of linked table when BE is in the same directory
- Solve bug in Imp/Exp specs export: Insert a null string instead of stopping in error when the spec has no description.
- Determine if Export or Import for any type of Export instead of only for text files.

### CODE REVIEW

**Comment by joyfullservice** [comment]
This looks great! Could you split out https://github.com/joyfullservice/msaccess-vcs-integration/pull/9/commits/fcb83ea76654a65e66280f590b54494142563724 into a separate pull request, since this is still under discussion? That would allow us to move forward with the export table fixes right away.

**Comment by A9G-Data-Droid** [comment]
I could. We are now running in to the problem with Access files in a version control system. If I have 2 PRs, it doesn't matter which order you accept them in the resulting .accda file will only contain the changes from the last merge. Each pull needs to be updated after each merge. Sigh. I should have just submitted these in the opposite order.

**Comment by joyfullservice** [comment]
I suppose if we got the build functions all working as intended, we could have even the add-in "build from source", which would make it much easier to implement pull requests and merges between multiple developers. Since the add-in is primarily VBA code, perhaps this wouldn't involve too many hours to get the basic functionality working... Hmm... Now just for a few hours to work on this...  :wink: 

It would be worth it for me to make this investment if we had a few of us that were interested in actively developing the add-in. Up to this point it has mainly just been me working on this branch for my own use, so that's why I hadn't spent much time on the GitHub collaboration aspects.

**Comment by A9G-Data-Droid** [comment]
So here's an idea: We could treat the .ACCDA as the "Release". Treat it like the binary that it is and don't check it in to Git at all. Then you can make GitHub Releases that only contain the two files the a user would download in a zip. This allows us to use GitHub as intended, and then you can release a new version whenever you see fit. Currently, the releases only have source code downloads. There should be the Binary release and the source code release, just like any other language. To make the binary release, you would just need to run "ImportAllSource" from the latest Master. That should replace all the modules with the source. If it doesn't then we need to fix that function.

**Comment by pmidhdev** [comment]
Hi Guys, 
I just read your comments of yesterday. Thanks for considering to pull my changes into the main branch, I feel very honored to be able to give any contribution :)
However is true that probably the code is redundant and difficult to follow, as @joyfullservice pointed out, because I added some functions I needed instead of changing some existing ones in order to be able to merge any eventual new version of master without to have to resolve too much conflicts. Of course please feel free to pick anything or comment it if I can help changing something.
As I about always worked with Access and VBA I have to confess that I'm new to version control in general and to Git/GitHub as well, so I don't understand the half of what you say, but I will and hopefully contribute to the project :)
In the last years I developed several applications and of course I got to the point to need both version control and use libraries where to group common functions so i looked around and met Git.
Despite I work alone and mainly with local Git repositories (but going to use company GitLab in the near future) I think I have similar needing: Serialize all FE objects for version control, and also some tables content (service and list tables), ribbon etc.. So for me RubberDuck wasn't enough as well, but it is useful for other things.
I think I tried the "ImportAllSource" function and saw that doesn't work so I'm still deploying the .accde version of the FE and to generate it I have to disconnect the VCS library.
I also checked the Application Starter @A9G-Data-Droid mentioned and seems very good, but for now I'll continue to use a DOS batch based common deploy as it happened in the past that in the company there were different schedules when deploying new versions of Access and I had to check and launch different versions of applications. The repair and compact function when nobody is connected seems very useful though I'm planning to migrate the BE to SQL server.
I'll keep in touch.
Ciao :)

**Comment by A9G-Data-Droid** [comment]
Ok, I have removed commit fcb83ea and mended the binary so that it matches the source. Verified by ExportAll. This is now just the TableFix so that each of my pull requests stand alone.

File: VCS_Loader.bas [hunk 1]
```diff
@@ -1,71 +1,71 @@
-Attribute VB_Name = "VCS_Loader"
```
Comment (A9G-Data-Droid) [inline VCS_Loader.bas]: I have no idea why it shows all lines changed. I deleted the file and then it came back with the revert. Maybe the line endings got flipped?

**Comment by A9G-Data-Droid** [comment]
@joyfullservice I made a project that can compile Access databases from source. This is an example of the things I am talking about. The binary is only found as a release. The source code contains only what is exported from your add-in. It's a mash up of your code and other branches. I think it might be nice to keep the import code separate from the Export code like this. The compiler is different than the version control system. However, they are related. I will need to push a change to your exporter to get property importing to work, for example.

Check it out here: https://github.com/A9G-Data-Droid/AccessDB-BuildFromSource

**Comment by joyfullservice** [comment]
@A9G-Data-Droid - I am excited to see the effort that is going in to this! The ability to build the database from source will be a very helpful prerequisite for utilizing a more standard multi-developer workflow where we can easily merge pull requests and contribute to the project. Keep up the good work!

**Pull request merged**